File: examples/ios/MobileTrackerExample/MobileTrackerExample/Models/ValidationResult.swift
import Foundation

/// Result of configuration validation
public enum ValidationResult {
    /// Configuration is valid
    case valid
    
    /// Configuration is invalid with error message
    case error(String)
    
    /// Check if validation passed
    public var isValid: Bool {
        switch self {
        case .valid:
            return true
        case .error:
            return false
        }
    }
    
    /// Get error message if validation failed
    public var errorMessage: String? {
        switch self {
        case .valid:
            return nil
        case .error(let message):
            return message
        }
    }
}

File: examples/android/src/main/java/ai/founderos/mobiletracker/example/ValidationResult.kt
package ai.founderos.mobiletracker.example

/**
 * Sealed class representing the result of validation
 */
sealed class ValidationResult {
    /**
     * Indicates that validation passed
     */
    object Valid : ValidationResult()

    /**
     * Indicates that validation failed with an error message
     * 
     * @property message The error message describing what validation failed
     */
    data class Error(val message: String) : ValidationResult()
}
